tests/modules/net_vpc_firewall/test_plan.py

```diff
@@ -12,27 +12,62 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
 def test_vpc_firewall_simple(plan_runner):
-  "Test vpc with no extra options."
+  "Test variable defaults."
   _, resources = plan_runner()
   assert len(resources) == 3
-  assert set([r['type'] for r in resources]) == set(
-      ['google_compute_firewall'])
-  assert set([r['values']['name'] for r in resources]) == set(
-      ['vpc-ingress-tag-http', 'vpc-ingress-tag-https', 'vpc-ingress-tag-ssh'])
-  assert set([r['values']['project'] for r in resources]) == set(['project'])
-  assert set([r['values']['network'] for r in resources]) == set(['vpc'])
+  assert set([r['type'] for r in resources]) == set(['google_compute_firewall'])
+  assert set([r['values']['name'] for r in resources]) == set([
+      'test-vpc-ingress-tag-http', 'test-vpc-ingress-tag-https',
+      'test-vpc-ingress-tag-ssh'
+  ])
+  assert set([r['values']['project'] for r in resources
+             ]) == set(['test-project'])
+  assert set([r['values']['network'] for r in resources]) == set(['test-vpc'])
+
+
+def test_vpc_firewall_rules(plan_runner):
+  "Test custom rules."
+  custom_rules = '''{
```

Comment:
you could conver this to yaml and then ensure both plans (from hcl and yaml) are the same. Validates that your factory code converts rules in the same way as variables.

Reply:
I see what you're doing here...

Reply:
see the org policies tests ;)